libs/iterator_support/include/hpx/iterator_support/traits/is_sentinel_for.hpp

```diff
@@ -0,0 +1,59 @@
+//  Copyright (c) 2020 Giannis Gonidelis
+//
+//  SPDX-License-Identifier: BSL-1.0
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#pragma once
+
+#include <hpx/iterator_support/traits/is_iterator.hpp>
+#include <hpx/type_support/always_void.hpp>
+
+#include <utility>
+
+namespace hpx { namespace traits {
+
+    template <typename Sent, typename Iter, typename Enable = void>
+    struct is_sentinel_for : std::false_type
+    {
+    };
+
+    template <typename Sent, typename Iter>
+    struct is_sentinel_for<Sent, Iter,
+        typename util::always_void<typename is_iterator<Iter>::type,
+            typename detail::equality_result<Iter, Sent>::type,
+            typename detail::equality_result<Sent, Iter>::type,
+            typename detail::inequality_result<Iter, Sent>::type,
+            typename detail::inequality_result<Sent, Iter>::type>::type>
+      : std::true_type
+    {
+    };
+
+#if defined(HPX_HAVE_CXX20_DISABLE_SIZED_SENTINEL_FOR)
+    template <typename Sent, typename Iter>
+    constexpr bool disable_sized_sentinel_for =
```

Comment:
On C++17 or above this variable could be defined `inline`. Please use `HPX_INLINE_CONSTEXPR_VARIABLE` instead of the `constexpr`, this will take care of the difference for different standard versions.

Reply:
OTOH, if `HPX_HAVE_CXX20_STD_DISABLE_SIZED_SENTINEL_FOR` is defined, we can probably assume that we're on C++20, so you could write `inline constexpr` to begin with. The `HPX_INLINE_CONSTEXPR_VARIABLE` would be needed for the `#else` below only.